botorch/sampling/pathwise/features/maps.py

```diff
@@ -105,29 +481,157 @@ def __init__(
+            kernel=kernel,
+            input_transform=input_transform,
+            output_transform=output_transform,
+        )
         self.register_buffer("weight", weight)
         self.register_buffer("bias", bias)
-        self.weight = weight
-        self.bias = bias
-        self.input_transform = input_transform
-        self.output_transform = output_transform
 
     def forward(self, x: Tensor) -> Tensor:
         out = x @ self.weight.transpose(-2, -1)
-        return out if self.bias is None else out + self.bias
+        return out if self.bias is None else out + self.bias.unsqueeze(-2)
 
     @property
-    def num_outputs(self) -> int:
-        if self.output_transform is None:
-            return self.weight.shape[-2]
+    def raw_output_shape(self) -> Size:
+        return self.weight.shape[-2:-1]
+
+
+class IndexKernelFeatureMap(KernelFeatureMap):
+    def __init__(
+        self,
+        kernel: kernels.IndexKernel,
+        input_transform: TInputTransform | None = None,
+        output_transform: TOutputTransform | None = None,
+        ignore_active_dims: bool = False,
+    ) -> None:
+        r"""Initializes an IndexKernelFeatureMap instance.
 
-        canary = torch.empty(
-            self.weight.shape[-2], device=self.weight.device, dtype=self.weight.dtype
+        Args:
+            kernel: IndexKernel whose features are to be returned.
+            input_transform: An optional input transform for the module.
+                For kernels with `active_dims`, defaults to a FeatureSelector
+                instance that extracts the relevant input features.
+            output_transform: An optional output transform for the module.
+            ignore_active_dims: Whether to ignore the kernel's active_dims.
+        """
+        if not isinstance(kernel, kernels.IndexKernel):
+            raise ValueError(f"Expected {kernels.IndexKernel}, but {type(kernel)=}.")
+
+        super().__init__(
+            kernel=kernel,
+            input_transform=input_transform,
+            output_transform=output_transform,
+            ignore_active_dims=ignore_active_dims,
         )
-        return self.output_transform(canary).shape[-1]
+
+    def forward(self, x: Tensor | None) -> LinearOperator:
+        if x is None:
+            return self.kernel.covar_matrix.cholesky()
+
+        i = x.long()
+        j = torch.arange(self.kernel.covar_factor.shape[-1], device=x.device)[..., None]
+        batch = torch.broadcast_shapes(self.batch_shape, i.shape[:-2], j.shape[:-2])
+        return InterpolatedLinearOperator(
+            base_linear_op=self.kernel.covar_matrix.cholesky(),
+            left_interp_indices=i.expand(batch + i.shape[-2:]),
+            right_interp_indices=j.expand(batch + j.shape[-2:]),
+        ).to_dense()
 
     @property
-    def batch_shape(self) -> Size:
-        return self.kernel.batch_shape
+    def raw_output_shape(self) -> Size:
+        return self.kernel.raw_var.shape[-1:]
```

Comment:
I could see the raw_var not always being on the `IndexKernel`, or have it be optional - we're looking into that ATM. Can you make it depend on the `covar_matrix` instead?